zadania_ruby/oop/products.rb

```diff
@@ -0,0 +1,84 @@
+class Products
+  def initialize
+    @products = []
+    load_products
+  end
+
+  def load_products
+    if @products == []
+      CSV.foreach('products.csv', headers: true) do |row|
+        @products << parse_line(row)
+      end
+    end
+    @products
+  end
+
+  def find(product)
+    result = nil
+    @products.each do |row|
+      if row[:name] == product
+        result = "#{row[:name]} #{row[:price]}#{row[:currency]}"
+      end
+    end
+    result
+  end
+
+  def find_cheaper_products(price)
+    @products.find_all do |row|
```

Comment:
Można to zrobić też w jednej pętli each. Ewentualnie rozbić to na dwie metody, jedną do wyszukiwannia i jedną do wyświetlania

Reply:
Tak robiliśmy na zajęciach :)